jobsdb/jobsdb.go

```diff
@@ -2144,60 +2118,49 @@ func (jd *HandleT) internalStoreWithRetryEachInTx(ctx context.Context, tx *sql.T
 	return
 }
 
+var ParameterFilters = []string{"destination_id"}
+
 // Creates a map of workspace:customVal:Params(Dest_type: []Dest_ids for brt and Dest_type: [] for rt)
 // and then loop over them to selectively clear cache instead of clearing the cache for the entire dataset
-func (jd *HandleT) populateCustomValParamMap(CVPMap map[string]map[string]map[string]struct{}, customVal string, params []byte, workspace string) {
-	if _, ok := CVPMap[workspace]; !ok {
-		CVPMap[workspace] = make(map[string]map[string]struct{})
+func (*HandleT) populateCustomValParamMap(CVPMap map[string]map[string]map[string]struct{}, job *JobT) {
+	if _, ok := CVPMap[job.WorkspaceId]; !ok {
+		CVPMap[job.WorkspaceId] = make(map[string]map[string]struct{})
+	}
+	if _, ok := CVPMap[job.WorkspaceId][job.CustomVal]; !ok {
+		CVPMap[job.WorkspaceId][job.CustomVal] = make(map[string]struct{})
 	}
-	if jd.queryFilterKeys.CustomVal {
-		if _, ok := CVPMap[workspace][customVal]; !ok {
-			CVPMap[workspace][customVal] = make(map[string]struct{})
-		}
 
-		if len(jd.queryFilterKeys.ParameterFilters) > 0 {
-			var vals []string
-			for _, key := range jd.queryFilterKeys.ParameterFilters {
-				val := gjson.GetBytes(params, key).String()
-				vals = append(vals, fmt.Sprintf("%s##%s", key, val))
-			}
-			key := strings.Join(vals, "::")
-			if _, ok := CVPMap[workspace][customVal][key]; !ok {
-				CVPMap[workspace][customVal][key] = struct{}{}
-			}
-		}
+	var vals []string
+	for _, key := range ParameterFilters {
+		val := gjson.GetBytes(job.Parameters, key).String()
+		vals = append(vals, fmt.Sprintf("%s##%s", key, val))
+	}
+
+	key := strings.Join(vals, "::")
+	if _, ok := CVPMap[job.WorkspaceId][job.CustomVal][key]; !ok {
+		CVPMap[job.WorkspaceId][job.CustomVal][key] = struct{}{}
```

Comment:
Since we are making changes here, it is an opportunity to introduce https://pkg.go.dev/expvar#Map, so we can inspect the contents of the map easily by hitting the debug/expvar endpoint.

Reply:
Not sure how easy this would be right? We would have to do insertions multiple times, and right now we expose this through rudder-cli already

Reply:
Can't think of an easy solution where these are updated to exp var and do not impact the performance of jobsdb